Day 4 - Marina

Marina7 opened the pull request:
both exercises

Commit:
Day 04, Exercise 01, containers

- make unit tests pass :)

Commit:
Day 4, Exercise 02, stl algorithms_test

Commit:
Day 4, Exercise 02, stl algorithms_test

- make unit tests pass

Commit:
Day 4, Exercise 02, stl algorithms_test

- improve difference test

ncaklovic commented:
Ponešto prijedloga:
- word_frequency::frequency pretražuje mapu dva puta
- možda bi is_vowel mogao prvo napraviti towlower pa onda imati 5 usporedbi umjesto 10
- ne treba lambda u 07b i 08b, samo pošalji is_vowel
- može li 09 bez reverse-a (ona 4 if-a se napišu u jednoj liniji sa `?:`)

Marina7 committed:
Day 4, fixes according to comments

- find and operator[] searched through map two times, use what find found if it did
- use towlower to avoid 10 comparisons
- 4 if statements replaced with ternary operator
- replace sort+reverse with sort using reverse iterator
- add another option with lambda for opposite comparison (if there is no overloaded operator<)

Commit:
Day 4 - don't use lambdas for is_prime and is_vowel
